libraries/botbuilder-dialogs-declarative/src/resources/fileResource.ts

```diff
@@ -6,41 +6,38 @@
  * Licensed under the MIT License.
  */
 
-import { IResource } from './resource';
-
 import fs = require('fs');
+import { Resource } from './resource';
 
-export class FileResource implements IResource {
+export class FileResource extends Resource {
 
-    private resourceId: string;
-    private path: string;
+    private _fullname: string;
 
+    /**
+     * Initialize a new instance of the `FileResouce` class.
+     * @param path path to file.
+     */
     public constructor(path: string) {
-        if (!path) {
-            throw new Error('path');
-        }
-
-        this.path = path;
-
+        super();
+        this._fullname = path;
         // The id will be the file name, without the path
-        this.resourceId = this.path.replace(/^.*[\\\/]/, '');
+        this._id = this._fullname.replace(/^.*[\\\/]/, '');
     }
 
+    /**
+     * The full path to the resource on disk
+     */
     public get fullName(): string {
-        return this.path;
-    }
-
-    public id(): string {
-        return this.resourceId;
+        return this._fullname;
     }
 
     public readText(): string {
-        const filePath = this.path;
-        const temp = fs.readFileSync(filePath, 'utf-8');
-        return temp;
+        const filePath = this._fullname;
+        const text = fs.readFileSync(filePath, 'utf-8');
+        return text;
     }
 
     public toString(): string {
-        return this.id();
+        return this._id;
```

Comment:
This change is technically parity with .NET, but why do we still have FileResource.toString() in .NET if this is just a function that does the equivalent of [`Resource.id`](https://github.com/microsoft/botbuilder-js/pull/2342/files#diff-b737d102b45d121bda767b2e389e8074R15)?

Reply:
> This change is technically parity with .NET, but why do we still have FileResource.toString() in .NET if this is just a function that does the equivalent of [`Resource.id`](https://github.com/microsoft/botbuilder-js/pull/2342/files#diff-b737d102b45d121bda767b2e389e8074R15)?

I'm not the one who designed this interface, I guess in C# every class can have its own ToString() method to help generate a string content representing itself, which might be useful in some cases, that is why(I guess) we can see there are ToString() methods everywhere. However, in JS we never use these methods to provide string contents for objects, so it is ok to remove them.